server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -697,8 +697,17 @@ public String getLastUploadedManifestFile() {
             return lastUploadedManifestFile;
         }
 
+        public ClusterMetadataManifest getLastAcceptedManifest() {
+            return lastAcceptedManifest;
+        }
+
         @Override
         public void setLastAcceptedState(ClusterState clusterState) {
+            // for non leader node, update the lastAcceptedClusterState
+            if (clusterState.getNodes().isLocalNodeElectedClusterManager() == false) {
+                lastAcceptedState = clusterState;
+                return;
+            }
```

Comment:
Does the leader node need to write the full cluster state on re-elections?

Reply:
That was the case this now, after this change only the diff will be uploaded from last accepted state.

Reply:
```
 if (clusterState.getNodes().isLocalNodeElectedClusterManager() == false) {
                lastAcceptedState = clusterState;
                return;
            }
```
Correct, But for leader this should return false and on node bounce case where all cluster manager are bounced `shouldWriteFullClusterState ` would return true since the `lastAcceptedManifest ` would return null?

Reply:
Correct, That would still be the case. If all the cluster manager nodes are bounced at once, we will not have any last accepted state or manifest in our memory, so we will be uploading the full state in that case.
Do you have any alternative approach in mind that would be better in this specific case here?

Reply:
Can we not get the latest manifest/state, just as we would download the latest state on quorum loss?

Reply:
So, basically on a Node restart get last published manifest/state and mark that as last accepted state for cluster manager nodes. I don't think that can create any issue, @soosinha what do you think about this?
If we are doing that, I would rather that that as a different PR.

Reply:
I'm good doing this incrementally

Reply:
When a node restarted, the cluster state is read from local and remote state is not downloaded.
If previous state/manifest is needed, then remote state has to be read irrespective of whether local state exists. This is a logic change and I would rather have it tracked as a separate issue.

Reply:
Created an issue #15691 to track this feature request.